### PR TITLE
add initial support of mxfp4

### PR DESCRIPTION
**Pull request by wenhuach21**
mainly for study

### CODE REVIEW

File: auto_round/data_type/mxfp.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    "mx_bfloat16": (8, 9, 127, 3.3895313892515355e+38, 1.1754943508222875e-38),
+    "mx_bf16": (8, 9, 127, 3.3895313892515355e+38, 1.1754943508222875e-38),
+}
+
+FP32_EXPONENT_BIAS = 127
+FP32_MIN_NORMAL = 2 ** (-FP32_EXPONENT_BIAS + 1)
+
+
+def quant_mx(tensor, bits, data_type, v, max_scale, **kwargs):
+    """Quantize the given tensor tensor using the specified parameters.
```
Comment (WeiweiZhang1) [inline auto_round/data_type/mxfp.py]: repeat word 'tensor'

File: auto_round/data_type/utils.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+def get_quant_func(dtype, bits, sym):
+    """Retrieve the quantization function based on data type, bit width, and symmetry.
+
+       This function returns the appropriate quantization function from the QUANT_FUNC_WITH_DTYPE
+       dictionary based on the provided data type (`dtype`), bit width (`bits`), and whether
+       the quantization is symmetric (`sym`). If the function does not exist, it asserts False.
+
+       Args:
+           dtype (str): The data type for the quantization (e.g., 'int8', 'float16').
```
Comment (WeiweiZhang1) [inline auto_round/data_type/utils.py]: e.g., 'int', 'mx_fp' and etc.

File: auto_round/data_type/utils.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+           bits (int): The bit width for the quantization (e.g., 8, 16).
```
Comment (WeiweiZhang1) [inline auto_round/data_type/utils.py]: e.g., 2, 3, 4, 8

File: auto_round/quantizer.py [hunk 4]
```diff
@@ -243,8 +100,10 @@ def __init__(self, orig_layer):
         self.orig_layer = orig_layer
 
     def forward(self, x):
-        x, _, _ = quant_tensor(x, self.orig_layer.act_bits, self.orig_layer.group_size, self.orig_layer.sym,
-                               scale_dtype=self.orig_layer.scale_dtype, q_scale_thresh=self.orig_layer.q_scale_thresh)
+        x, _, _ = quant_tensor(self.orig_layer.act_quant_func, x, self.orig_layer.act_bits, self.orig_layer.group_size,
+                               self.orig_layer.sym,
```
Comment (WeiweiZhang1) [inline auto_round/quantizer.py]: could remove self.orig_layer.sym?

**Review by WeiweiZhang1 — APPROVED** [review]

File: auto_round/quantizer.py (hunk 4 above)
Comment (wenhuach21) [reply auto_round/quantizer.py]: nice catch, it should be a bug

**Pull request merged**